### PR TITLE
table calendar starting day of week

### LINKED ISSUE #22: [Schedule] Allow changing starting day of week
**Describe the solution you'd like**
As a user, I would like to change the starting day of the week.

**Additional context**
Currently, the calendar package we are using doesn't support the change of startingDayOfWeek at runtime.
Will be resolved when: https://github.com/aleksanderwozniak/table_calendar/issues/164

### LINKED ISSUE #92: [Schedule] Can change the calendar format and settings aren't updated.
**Describe the bug**
On the schedule view, we can drag down/up the calendar to change the format (week <-> 2 weeks <-> month) without going into the settings (see video). This change doesn't update the settings.

**To Reproduce**
1. Go to 'Schedule'
2. Drag down the calendar (the format should become 2 weeks instead of one week)
3. Go see the schedule settings
4. The calendar format shouldn't have changed

**Expected behavior**
The settings should reflect what we see on screen, except if we want this to become a feature.
Something like: "as a user, if I want to quickly see the 2 weeks ahead but don't want this to become my default view" :thinking: 

**Screenshots**

https://user-images.githubusercontent.com/22211097/114116168-2c506900-98b2-11eb-9ece-9287cc88eccb.mp4

### PR DESCRIPTION
**Pull request by MysticFragilist**
### Table calendar

This PR principally focus on upgrading the table_calendar package to version 3.0.0 to be able to set a starting day on the calendar. Thanks to @apomalyn, this was a lot more easy than expected 😄. The PR will allow #68 to be available to work on. Some other packages are also upgraded in forecast of null safety. 

I manage to change the settings for the number of weeks shown when the calendar is change by dragging the week up or down.

image
image

### Animation weird move

I also ran the `--cache-sksl` command to generate a file (script/flutter_01.sksl.json) to be used when building the app to avoid older hardware some lagging. (this did not fix the issue #85)

Also updated the wiki accordingly.

closes #22 
closes #92

### CODE REVIEW

File: lib/core/constants/markers.dart [hunk 1]
```diff
@@ -2,19 +2,19 @@ import 'package:google_maps_flutter/google_maps_flutter.dart';
...
+      position: const LatLng(45.495089693692194, -73.56374294991838),
+      markerId: const MarkerId('security_building_B'),
+      infoWindow: InfoWindow(title: intl.security_station),
+    ),
+    Marker(
+      position: const LatLng(45.49391646843658, -73.5634878349083),
+      markerId: const MarkerId('security_building_D'),
+      infoWindow: InfoWindow(title: intl.security_station),
+    ),
+  ];
```
Comment (AlexandreMarkus) [inline lib/core/constants/markers.dart:20]: If there is no changes made in this file remove it from the PR

Comment (MysticFragilist) [reply lib/core/constants/markers.dart:20]: I added the `const` keyword over the MarkerId, it seems that before it wasn't require but now for some reason I got some warning about it. Don't know if on intellij it's doing the same thing though. What do you think?
image

Comment (AlexandreMarkus) [reply lib/core/constants/markers.dart:20]: Ohh great, I didn't pay attention to the word const. Keep it for sure!!

**Comment by AlexandreMarkus** [comment]
@MysticFragilist tests are not passing

**Review by camillebrulotte — COMMENTED** [review]
Good job ! Seems to work fine on IOS too :D

**Comment by AlexandreMarkus** [comment]
Do current day is working on IOS?

**Comment by camillebrulotte** [comment]
> Do current day works on IOS?

En retestant je crois qu'il manque quelque chose ! Si je suis sur le même mois que "aujourd'hui" ça fonctionne. MAIS si je suis sur un autre mois, le calendar ne me ramène pas à aujourd'hui (malgré que le carrée rouge revient à aujourd'hui). Je ne sais pas si ça fait pareille sur Android mais probablement ? @MysticFragilist Si c'est possible normalement le calendrier devrait te ramener au bon mois aussi ! 

https://user-images.githubusercontent.com/37625944/117575351-62e6f280-b0af-11eb-92e4-c0fdcecdf49d.mov

**Comment by MysticFragilist** [comment]
good catch! 😄  je vais essayer de regarder ça en fin de semaine!

**Comment by AlexandreMarkus** [comment]
> good catch! 😄  je vais essayer de regarder ça en fin de semaine! Also, régler mes tests, parce que depuis le passage à 3.0.0, il semble que le findByText ne trouve plus les carrées des jours (I.E. findByText("26") par exemple lance une exception, bien qu'il soit display on screen ou offstage), je vais devoir trouver un workaround  pour ça.

Je crois que c'est en lien avec le _calendarController. Sur Android aussi le bouton currentDay ne semble pas fonctionner

**Comment by AlexandreMarkus** [comment]
@MysticFragilist @camillebrulotte fix is done for today's button!!

**Comment by MysticFragilist** [comment]
Nice fix! Well done and thanks again! 😄 I'm currently waiting for an answer from  [this issue](https://github.com/aleksanderwozniak/table_calendar/issues/505), once I manage to figure out the test issue, I'll be back to close once and for all this PR. 😃

**Comment by camillebrulotte** [comment]
> @MysticFragilist @camillebrulotte fix is done for today's button!!
@AlexandreMarkus 
good job ! Works on IOS too ! :D

**Comment by AlexandreMarkus** [comment]
image
Lorsque je clique sur Horaire, il ouvre sur la mauvaise date. Par contre le bouton Today fonctionne bien

File: lib/ui/router.dart [hunk 2]
```diff
@@ -43,22 +43,25 @@ class AppRouter {
       case RouterPaths.schedule:
         return PageRouteBuilder(
             settings: RouteSettings(name: routeSettings.name),
-            pageBuilder: (_, __, ___) => const ScheduleView());
+            pageBuilder: (_, __, ___) =>
+                ScheduleView(initialDay: DateTime(2020, 2, 15)));
       case RouterPaths.student:
         return PageRouteBuilder(
             settings: RouteSettings(name: routeSettings.name),
```
Comment (AlexandreMarkus) [inline lib/ui/router.dart:50]: Je crois qu'il faudrait mettre DateTime.now()

**Review by AlexandreMarkus — CHANGES_REQUESTED** [review]

**Comment by AlexandreMarkus** [comment]
test/ui/views/goldenFiles/scheduleView_5.png ce golden semble avoir trop de différence avec le précédent. On dirait que ce n'est plus la même date.

File: lib/ui/router.dart (hunk 2 above)
Comment (MysticFragilist) [reply lib/ui/router.dart:50]: Bien vu! j'avais fait ça pour tester quelque chose, je vais revert

**Comment by MysticFragilist** [comment]
@AlexandreMarkus, when you will have time, could you launch `flutter test`? It seems that the github actions detect a mismatch on the golden aboutview_1.png, but on my local machine all tests passed which is weird... Does it fail on yours?
Screenshot from 2021-05-22 18-29-00

**Comment by AlexandreMarkus** [comment]
> @AlexandreMarkus, when you will have time, could you launch `flutter test`? It seems that the github actions detect a mismatch on the golden aboutview_1.png, but on my local machine all tests passed which is weird... Does it fail on yours?
> Screenshot from 2021-05-22 18-29-00
> 

I don't know why but since a few days, this golden has problems sometimes but restarting the workflow again seems to work for now

**Review by AlexandreMarkus — APPROVED** [review]
!!!

**Review by camillebrulotte — APPROVED** [review]
Niice tout semble bien fonctionner encore sur IOS aussi  :D

**Pull request merged**